haystack/document_stores/elasticsearch.py

```diff
@@ -1861,74 +1867,79 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
-                # bad embedding field
-                if mappings["properties"][self.embedding_field]["type"] != "knn_vector":
-                    raise Exception(
-                        f"The '{index_name}' index in OpenSearch already has a field called '{self.embedding_field}'"
-                        f" with the type '{mappings['properties'][self.embedding_field]['type']}'. Please update the "
-                        f"document_store to use a different name for the embedding_field parameter."
+            indices = self.client.indices.get(index_name, headers=headers)
+            for index_id, index_info in indices.items():
+                mappings = index_info["mappings"]
+                index_settings = index_info["settings"]["index"]
+                if self.search_fields:
+                    for search_field in self.search_fields:
+                        if (
+                            search_field in mappings["properties"]
+                            and mappings["properties"][search_field]["type"] != "text"
+                        ):
+                            raise Exception(
+                                f"The search_field '{search_field}' of index '{index_id}' with type '{mappings['properties'][search_field]['type']}' "
+                                f"does not have the right type 'text' to be queried in fulltext search. Please use only 'text' type properties as search_fields or use another index. "
+                                f"This error might occur if you are trying to use haystack 1.0 and above with an existing elasticsearch index created with a previous version of haystack. "
+                                f'In this case deleting the index with `delete_index(index="{index_id}")` will fix your environment. '
+                                f"Note, that all data stored in the index will be lost!"
+                            )
+
+                # embedding field will be created
+                if self.embedding_field not in mappings["properties"]:
+                    mappings["properties"][self.embedding_field] = self._get_embedding_field_mapping(
+                        similarity=self.similarity
                     )
-                # embedding field with global space_type setting
-                if "method" not in mappings["properties"][self.embedding_field]:
-                    embedding_field_space_type = index_settings["knn.space_type"]
-                # embedding field with local space_type setting
+                    self.client.indices.put_mapping(index=index_id, body=mappings, headers=headers)
+                    self.embeddings_field_supports_similarity = True
                 else:
+                    # bad embedding field
+                    if mappings["properties"][self.embedding_field]["type"] != "knn_vector":
+                        raise Exception(
+                            f"The '{index}' index in OpenSearch already has a field called '{self.embedding_field}'"
```

Comment:
@ZeJ0hn `index` is not defined here as I found out from the failing test. I think it makes sense to use `index_id` and maybe also `index_name` here, right?

Reply:
Already fixed. Wow. 🤩

Reply:
My first pull request in your project so I need to be reactive! :nerd_face: